### ISSUE TITLE
[Question] Correct Alphafold requirements?

### ISSUE THREAD

**Issue by ccoulombe** [deepmind/alphafold#511]
Hello,

I'm looking to clarify what are the correct python requirements for Alphafold.

The [`requirements.txt`](https://github.com/deepmind/alphafold/blob/main/requirements.txt) list a bunch of hard-pinned python packages while the `setup.py` does not. In other words, the [`install_requires`](https://github.com/deepmind/alphafold/blob/b0b7afadbdc079a0b48d49a3d54eb0ca973a2da1/setup.py#L30) accepts any version of Numpy while the requirements file strickly ask for the `1.19.5`. This also raises, does Alphafold strictly works with TF 2.5 or is it compatible with 2.9 for instance?

Which one is correct? 

Thanks for any insights.

**Comment by ccoulombe** [comment]
@Augustin-Zidek might have the answer since he's the committer for the `setup.py` ? Thanks in advance!

**Comment by tomwardio** [comment]
Hi @ccoulombe,

The `requirements.txt` is the canonical package versions that we've tested works with AFOS. The `setup.py` is what we use to test newer versions and run our test suite.

Whilst it's likely that AFOS would work with newer versions of various packages (e.g. TensorFlow), it's not something that we've validated, so we'd recommend using the `requirements.txt` (or even better, Docker/colab notebook) to ensure results are accurate.

Hope this helps!
